### PR TITLE
Adding Missouri pre migration work

### PR DESCRIPTION
**Pull request by JessicaWNava**
This PR implements the pre migration work for MO. It is based off of Indiana because it needs the changes to the simple report transform, but will be based off master once that PR is merged.
Test Steps:
1. Review that the changes in organizations.yml match MO previous settings
2. Check that the resulting message matches the outbound message from the Covid pipeline
3. Make sure the tests pass

## Changes
- Added settings for MO to organizations.yml
- Created the MO receiver transform
- Added an integration test for MO

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MO/MO-receiver-transform.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+hl7Class: ca.uhn.hl7v2.model.v251.message.ORU_R01
+
+extends: classpath:/metadata/hl7_mapping/ORU_R01/ORU_R01-base.yml
```
Comment (oslynn) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MO/MO-receiver-transform.yml:3]: You can add the following constants and use them later in this transform.

constants:
  PID: '/PATIENT_RESULT/PATIENT/PID'
  ORC: '/PATIENT_RESULT/ORDER_OBSERVATION/ORC'
  OBR: '/PATIENT_RESULT/ORDER_OBSERVATION/OBR'

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MO/MO-receiver-transform.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+elements:
+  - name: mo-patient-race-coding-system
+    resource: 'Bundle.entry.resource.ofType(Patient).extension("http://ibm.com/fhir/cdm/StructureDefinition/local-race-cd").value.coding'
+    condition: '%resource.code.exists()'
+    value: [ '%resource.system.getCodingSystemMapping()' ]
+    hl7Spec: [ '/PATIENT_RESULT/PATIENT/PID-10-3' ]
```
Comment (oslynn) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MO/MO-receiver-transform.yml:10]: Now you can set hl7Spec: [ '%{PID}-10-3' ]

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MO/MO-receiver-transform.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  - name: mo-patient-race-coding-system
```
Comment (oslynn) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MO/MO-receiver-transform.yml:12]: Please delete this extra element.

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MO/MO-receiver-transform.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    value: [ '"MODHSS"' ]
+    hl7Spec: [ 'MSH-6-1' ]
+
+  - name: mo-receiving-application-universal-Id
+    value: [ '"2.16.840.1.114222.4.3.2.2.1.163"' ]
+    hl7Spec: [ 'MSH-6-2' ]
+
+  - name: mo-receiving-application-universal-id-type
+    value: [ '"ISO"' ]
+    hl7Spec: [ 'MSH-6-3' ]
```
Comment (oslynn) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MO/MO-receiver-transform.yml:40]: According to the setting above (suppressHl7Fields: "MSH-19-3, PID-15-3, ORC-14-1, OBR-17-1"), you can suppress them here as following:

  - name: mo-unwanted-fields
    condition: 'true'
    value: [ '""' ]
    hl7Spec: [ 'MSH-19-3', '%{PID}-13-12', '%{PID}-15-3', '%{ORC}-14-1', '%{OBR}-17-1'  ]

File: prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/sample_MO_20240628-0001.fhir [hunk 5]
```diff
@@ -0,0 +1 @@
...r":[{"use":"official","type":{"coding":[{"system":"http://loinc.org","code":"81959-9","display":"Public health laboratory ask at order entry panel"}]}}],"status":"final","code":{"coding":[{"system":"http://loinc.org","code":"95418-0","display":"Employed in a healthcare setting"}],"text":"Employed in a healthcare setting"},"subject":{"reference":"Patient/7c0d1de9-270e-4d9c-a4ec-af92560cec67"},"valueCodeableConcept":{"coding":[{"system":"http://terminology.hl7.org/ValueSet/v2-0136","code":"Y","display":"Yes"}]}}},{"fullUrl":"Observation/0042a1cb-8473-3181-9b65-fcb08dc112a3","resource":{"resourceType":"Observation","id":"0042a1cb-8473-3181-9b65-fcb08dc112a3","identifier":[{"use":"official","type":{"coding":[{"system":"http://loinc.org","code":"81959-9","display":"Public health laboratory ask at order entry panel"}]}}],"status":"final","code":{"coding":[{"system":"http://loinc.org","code":"95421-4","display":"Resides in a congregate care setting"}],"text":"Resides in a congregate care setting"},"subject":{"reference":"Patient/7c0d1de9-270e-4d9c-a4ec-af92560cec67"},"valueCodeableConcept":{"coding":[{"system":"http://terminology.hl7.org/ValueSet/v2-0136","code":"Y","display":"Yes"}]}}}]}
```
Comment (oslynn) [inline prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/sample_MO_20240628-0001.fhir:1]: Perhaps, you can fix the integration test, if needed.

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MO/MO-receiver-transform.yml (hunk 1 above)
Comment (JessicaWNava) [reply prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/MO/MO-receiver-transform.yml:3]: Only one of those is applicable, but I can add it.

**Review by oslynn — APPROVED** [review]

**Comment by oslynn** [comment]
Looking good to me.

File: prime-router/settings/STLTs/MO/mo-phd.yml [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+            - "ORC-7-10-5"
+            - "OBR-2-2"
+            - "OBR-3-2"
+            - "SPM-2-1-2"
+            - "SPM-2-2-2"
+            - "ORC-21-1"
+            - "OBX-23-1"
+            - "BX-15.1"
+            - "OBX-23.10"
+          customLengthHl7Fields: {}
```
Comment (victor-chaparro) [inline prime-router/settings/STLTs/MO/mo-phd.yml:72]: These should be:
```
  - "OBX-15-1"
  - "OBX-23-10"
```
This is going to be tedious, but can you check these fields are getting truncated by adding an integration test. We've had issue with truncation before and some fields were not getting truncated depending on where they were on the message. Specially these fields:
```
- "SPM-2-1-2"
- "SPM-2-2-2"
- "ORC-7-10-5"
```
I don't know if the syntax for truncating them in the UP is the right one. We haven't had to truncate fields in the UP that are that nested.

**Review by victor-chaparro — CHANGES_REQUESTED** [review]

File: prime-router/settings/STLTs/MO/mo-phd.yml (hunk 6 above)
Comment (JessicaWNava) [reply prime-router/settings/STLTs/MO/mo-phd.yml:72]: Truncate usually means shorten, so what does truncate mean in this case?

Comment (JessicaWNava) [reply prime-router/settings/STLTs/MO/mo-phd.yml:72]: Truncation does not work properly. I created two tickets for this: https://app.zenhub.com/workspaces/engagement-6166edbd409257001e09f1b2/issues/gh/cdcgov/prime-reportstream/15110 and https://app.zenhub.com/workspaces/engagement-6166edbd409257001e09f1b2/issues/gh/cdcgov/prime-reportstream/15109 
can we merge this PR and go back to address this when these are fixed? @victor-chaparro

**Review by victor-chaparro — APPROVED** [review]
I'll approve it, but can you update the ones that are incorrect even if they're not working.

**Comment by JessicaWNava** [comment]
> I'll approve it, but can you update the ones that are incorrect even if they're not working.

Yep, done

**Pull request merged**